libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -90,7 +97,11 @@ public void onDestroy() {
   public void onProgressChange(Location location, RouteProgress routeProgress) {
     this.routeProgress = routeProgress;
     instructionModel.setValue(new InstructionModel(getApplication(), routeProgress, locale, unitType));
-    summaryModel.setValue(new SummaryModel(getApplication(), routeProgress, locale, unitType));
+    Calendar time = Calendar.getInstance();
+    boolean isDeviceTwentyFourHourFormat = DateFormat.is24HourFormat(getApplication());
+    String arrivalTime = formatTime(time, (int) routeProgress.durationRemaining(), timeFormatType,
+      isDeviceTwentyFourHourFormat);
+    summaryModel.setValue(new SummaryModel(getApplication(), routeProgress, locale, unitType, arrivalTime));
```

Comment:
Would it make sense to move the arrival time formatting into the `SummaryModel` itself?  `Application` and `routeProgress` are already being passed.  I think it would clean up `onProgressChange` 

It would also remove the duplicate code found in `SummaryBottomSheet`